apps/roofer-app/roofer-app.cpp

```diff
@@ -512,185 +603,331 @@ int main(int argc, const char* argv[]) {
     auto tile_extents =
         create_tiles(roi, roofer_cfg.tilesize_x, roofer_cfg.tilesize_y);
 
-    for (auto& tile : tile_extents) {
-      auto& building_tile = building_tiles.emplace_back();
+    for (std::size_t tid = 0; tid < tile_extents.size(); tid++) {
+      auto& tile = tile_extents[tid];
+      auto& building_tile = initial_tiles.emplace_back();
+      building_tile.id = tid;
       building_tile.extent = tile;
       building_tile.proj_helper = roofer::misc::createProjHelper();
     }
   }
-  logger.debug("Created {} batch tile regions", building_tiles.size());
+  logger.debug("Created {} batch tile regions", initial_tiles.size());
 
   // Multithreading setup
   size_t nthreads = std::thread::hardware_concurrency();
-  // -3, because we need one thread for crop, reconstruct, serialize each
-  lf::lazy_pool pool(nthreads - 3);
+  // -5, because we need one thread for crop, reconstruct, sort, serialize,
+  // plus logger. We don't count with the main thread and tracer thread, because
+  // all the work is offloaded to the worker threads and the main is not doing
+  // much work, tracer either.
+  size_t nthreads_reserved = 5;
+  size_t jobs_from_param = 0;
+  if (cmdl({"-j", "--jobs"}) >> jobs_from_param) {
+    // Limit the parallelism
+    if (jobs_from_param < nthreads_reserved + 1) {
+      // Only one thread will be available for the reconstructor pool
+      nthreads = nthreads_reserved + 1;
+    } else {
+      nthreads = jobs_from_param;
+    }
+  }
+  size_t nthreads_reconstructor_pool = nthreads - nthreads_reserved;
```

Comment:
this could become negative if nthreads <  nthreads_reserved and no -j was specified